src/main/kotlin/org/rust/ide/refactoring/extractFunction/RsExtractFunctionHandler.kt

```diff
@@ -75,6 +71,32 @@ class RsExtractFunctionHandler : RefactoringActionHandler {
         }
     }
 
+    /**
+     * Finds inherent impl corresponding to [traitImpl].
+     * Impls at same tree level are checked (e.g. if [traitImpl] is top-level impl, then top-level impls are checked).
+     */
+    private fun findExistingInherentImpl(traitImpl: RsImplItem): RsImplItem? {
+        check(traitImpl.traitRef != null)
+        val traitCachedImpl = RsCachedImplItem.forImpl(traitImpl)
+        return (traitImpl.parent as? RsItemsOwner)
+            ?.childrenOfType<RsImplItem>()
+            ?.firstOrNull { impl ->
+                val cachedImpl = RsCachedImplItem.forImpl(impl)
+                cachedImpl.isInherent && cachedImpl.isValid
+                    && cachedImpl.typeAndGenerics == traitCachedImpl.typeAndGenerics
```

Comment:
This won't work with generic impls, is it intended? If yes, maybe we could express it more explicitly, i.e. skip impls with generics in this condition?

Reply:
Generics are currently not supported when new inherent impl is created (#7801), so yes, it is expected. I added explicit check